### PR TITLE
Add Elixir docs

### PR DESCRIPTION
**Pull request by roytomeij**
Move some of the Elixir documentation from Hexdocs to docs.appsignal.com, including an updated explanation on why we decided to use NIF's (this needs to be checked for technical inconsistencies).

Todo:

* [x] Add content to the "overview" page.

First setup for #69

### CODE REVIEW

**Comment by roytomeij** [comment]
Please fix what's wrong, and try to stick to acknowledging that NIF's could be a bad idea but not in this case, and focus on customer benefits (battle tested among all customer).

**Comment by thijsc** [comment]
Improved the NIF explanation.

File: source/elixir/why-nif.html.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+title: "Why a NIF?"
+---
+
+The AppSignal for Elixir integration relies on a NIF that calls out to a library that's written in [Rust](https://www.rust-lang.org) to do the heavy lifting. It's job is to receive data from the Elixir side, normalize and write it to a separate agent process over a Unix socket. The agent then sends the data to AppSignal.com periodically for further processing.
+
+While using NIF's has inherent risks, using one in this case has several advantages:
```
Comment (wesoudshoorn) [inline source/elixir/why-nif.html.md]: "has inherent risks" -> can we add a section where we explicitly state these risks and what we did to minimize them? In what scenarios might a NIF become a problem and how do we handle those situation witch grace?

**Comment by tombruijn** [comment]
I've updated the Ruby overview page to give a small intro to the different subjects of the Ruby documentation.

https://github.com/appsignal/appsignal-docs/compare/55e755179a022988e0b56afecdbd16e2394e3d48...602ce463f208ca623c125cee3736d8813be1488f

Maybe we can do the same for Elixir?

File: source/elixir/why-nif.html.md (hunk 1 above)
Comment (roytomeij) [inline source/elixir/why-nif.html.md]: That's a good one Wes. I addressed how me minimize those risks, but I didn't include what they are :-)

**Comment by roytomeij** [comment]
We do need to add content to the `elixir/index` page, but I'm not sure what.

**Comment by tombruijn** [comment]
I added a big ol' update. Split up pages and imported some more documentation from the hexdocs. Restructured it to match the Ruby docs.

@jeffkreeftmeijer Could you add an example to the Elixir decorator instrumentation in `elixir/instrumentation/index.html.md`? (I don't know how that works 🙊 )
That's the only thing missing now, I think.

**Pull request merged**